dnf-json

```diff
@@ -248,159 +178,144 @@ class Solver():
-                                )
-                            )
-                            log.info("depsolve success")
-
-                    except dnf.exceptions.MarkingErrors as e:
-                        log.info("error install_specs")
-                        self.response_with_dnf_error(
-                            "MarkingErrors",
-                            f"Error occurred when marking packages for installation: {e}"
-                        )
-                    except dnf.exceptions.DepsolveError as e:
-                        log.info("error depsolve")
-                        self.response_with_dnf_error(
-                            "DepsolveError",
-                            (
-                                "There was a problem depsolving "
-                                f"{arguments['package-specs']}: {e}"
-                            )
-                        )
-                    except dnf.exceptions.Error as e:
-                        self.response_with_dnf_error(
-                            type(e).__name__,
-                            f"Error occurred when setting up repo: {e}")
-            finally:
-                for cache_folder in self.init_cache_folder_list(repos):
-                    cache_state.update_used(cache_folder)
-                cache_state.clean_unused()
-                cache_state.store()
-
-
-log.info("Starting the dnf-json server")
-
-LISTEN_FDS = int(os.environ.get("LISTEN_FDS", 0))
-# set from entrypoint if differs from 3
-LISTEN_FD = int(os.environ.get("LISTEN_FD", 3))
-
-
-# The dnf-json web server has to use forks to serve the requests. Because the
-# dnf library is leaking memory in its Cpp side.
-class SystemDActivationSocketServer(socketserver.ForkingMixIn, socketserver.UnixStreamServer):
-    def server_bind(self):
-        log.debug("service bind")
-        log.debug("rebind socket")
-        log.debug("address_family: %d ", self.address_family)
-        log.debug("socket_type: %d ", self.socket_type)
-        if LISTEN_FDS > 1:
-            log.warning("More than one LISTEN_FDS")
-        self.socket = socket.fromfd(LISTEN_FD, self.address_family, self.socket_type)
-
-
-# start the web server
-server = SystemDActivationSocketServer('', DnfJsonRequestHandler)
-server.serve_forever()
+                solver = Solver(
+                    repos,
+                    module_platform_id,
+                    persistdir,
+                    cache_dir,
+                    arch
+                )
+                if command == "dump":
+                    result = solver.dump()
+                    printe("dump success")
+                elif command == "depsolve":
+                    result = solver.depsolve(
+                        job["package-specs"],
+                        job.get("exclude-specs", [])
+                    )
+                    printe("depsolve success")
+                results.append(result)
```

Comment:
Thanks for reminding me of this.  From what I understand, when we do the chained depsoving, we'll probably only want to return the last depsolve result or merge them.  Either way, we should get it right here once we know how it will work.